### PR TITLE
[leminv] [declare] Use higher-level Declare API.

### PR DESCRIPTION
**Pull request by ejgallego**
Nit on using the higher-level API + bugfix for Derive API.

### CODE REVIEW

File: plugins/ltac/leminv.ml [hunk 1]
```diff
@@ -246,11 +247,7 @@ let add_inversion_lemma_exn ~poly na com comsort bool tac =
   let sigma = Evd.from_env env in
   let sigma, c = Constrintern.interp_type_evars ~program_mode:false env sigma com in
   let sigma, sort = Evd.fresh_sort_in_family ~rigid:univ_rigid sigma comsort in
-  try
-    add_inversion_lemma ~poly na env sigma c sort bool tac
-  with
-    |   UserError (Some "Case analysis",s) -> (* Reference to Indrec *)
-          user_err ~hdr:"Inv needs Nodep Prop Set" s
```
Comment (ejgallego) [inline plugins/ltac/leminv.ml:253]: Note: this was dead code.

**Comment by SkySkimmer** [comment]
check_evars_are_solved is too strict, we only care about the evars in the current definition.

**Comment by SkySkimmer** [comment]
May work after https://github.com/coq/coq/pull/12759, at least the test for #2775 is fixed when I tried merging the 2 PRs.

**Comment by ejgallego** [comment]
> May work after #12759, at least the test for #2775 is fixed when I tried merging the 2 PRs.

Let's see :)

File: vernac/declare.ml [hunk 2]
```diff
@@ -2014,7 +2014,7 @@ let finish_derived ~f ~name ~entries =
   let lemma_def = Internal.map_entry_body lemma_def ~f:(fun ((b,ctx),fx) -> (substf b, ctx), fx) in
   let lemma_def = DefinitionEntry lemma_def in
   let ct = declare_constant ~name ~kind:Decls.(IsProof Proposition) lemma_def in
-  [GlobRef.ConstRef ct]
+  [GlobRef.ConstRef f_kn; GlobRef.ConstRef ct]
```
Comment (SkySkimmer) [inline vernac/declare.ml:2017]: is this related to leminv?

Comment (ejgallego) [reply vernac/declare.ml]: Ups, fixed.

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
Anybody understanding the errors spitted by coq_tools?

**Comment by ejgallego** [comment]
> Anybody understanding the errors spitted by coq_tools?

Spurious / unrelated so far, let's rerun just in case.
